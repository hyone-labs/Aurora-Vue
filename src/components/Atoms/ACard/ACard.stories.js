import { ACard } from '.';

export default {
  title: 'Design System/Atoms/ACard',
}

export const withText = () => ({
  components: { ACard },
  template: '<a-card icon="fas fa-user">Hello Button</a-card>',
})

export const withElevation = () => ({
  render() {
    return (
      <div>
        <ACard elevation="low">With JSX</ACard>
        <p />
        <ACard>With JSX</ACard>
        <p />
        <ACard elevation="high">With JSX</ACard>
      </div>
    );
  }
})

export const withVariantInfo = () => ({
  render() {
    return (
      <div>
        <ACard elevation="low">With JSX</ACard>
        <p />
        <ACard elevation="low" variant="success">With JSX</ACard>
        <p />
        <ACard elevation="low" variant="info">With JSX</ACard>
        <p />
        <ACard elevation="low" variant="danger">With JSX</ACard>
        <p />
        <ACard elevation="low" variant="warn">With JSX</ACard>
        <p />
        <ACard elevation="low" variant="dark">With JSX</ACard>
      </div>
    );
  }
})