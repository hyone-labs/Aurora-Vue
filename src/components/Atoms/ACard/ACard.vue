<template>
  <component
    :class="{
      [`a-card`]: true,
      [`a-card--${variant}`]: variant,
      [`a-card--${size}`]: size,
      [`a-card--elevation-${elevation}`]: elevation,
    }"
    :is="tag"
  >
    <slot />
  </component>
</template>

<script>
export default {
  props: {
    variant: {
      type: String,
      default: '',
    },
    size: {
      type: String,
      default: '',
    },
    elevation: {
      type: String,
      default: 'medium',
    },
    tag: {
      type: String,
      default: 'div',
    },
  },
};
</script>

<style lang="scss">
.a-card {
  background-color: var(--colors-original-white);
  border-radius: 10px;
  padding: 24px;
  position: relative;

  &--elevation {
    &-low {
      box-shadow: var(--shadow-elevation-01);
    }

    &-medium {
      box-shadow: var(--shadow-elevation-02);
    }

    &-high {
      box-shadow: var(--shadow-elevation-03);
    }
  }
}
</style>
