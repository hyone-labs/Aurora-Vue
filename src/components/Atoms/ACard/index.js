import ACard from './ACard.vue';

export { ACard };