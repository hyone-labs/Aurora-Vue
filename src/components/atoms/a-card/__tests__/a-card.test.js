import { render } from '@testing-library/vue';
import ACard from '../a-card.vue';

describe('ACard', () => {
  it('render correctly with default props', () => {
    const { html } = render(ACard, {
      slots: {
        default: 'My first test',
      },
    });
    expect(html()).toMatchSnapshot();
  });

  it('render correctly with variant', () => {
    const { container } = render(ACard, {
      props: {
        variant: 'primary',
      },
      slots: {
        default: 'My second test',
      },
    });
    expect(container.firstChild.classList.contains('a-card--variant-primary')).toBe(true);
  });

  it('render correctly with elevation', () => {
    const { container } = render(ACard, {
      props: {
        elevation: 'high',
      },
      slots: {
        default: 'My second test',
      },
    });
    expect(container.firstChild.classList.contains('a-card--elevation-high')).toBe(true);
  });

  it('render correctly with elevation', () => {
    const { container } = render(ACard, {
      props: {
        tag: 'section',
      },
      slots: {
        default: 'My second test',
      },
    });
    expect(container.firstChild.tagName).toBe('SECTION');
  });
});
