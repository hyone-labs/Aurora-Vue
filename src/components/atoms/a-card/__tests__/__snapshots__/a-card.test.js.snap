// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACard render correctly with default props 1`] = `"<div class=\\"a-card a-card--variant-default a-card--elevation-medium\\">My first test</div>"`;
